cmd/copy.go

```diff
@@ -948,6 +1025,136 @@ func (cca *cookedCopyCmdArgs) processCopyJobPartOrders() (err error) {
 	return nil
 }
 
+// TODO: include this within initResourceTraverser?
+func initPipeline(ctx context.Context, location common.Location, credential common.CredentialInfo) (p pipeline.Pipeline, err error) {
+	switch location {
+	case common.ELocation.Local():
+		return nil, nil
+	case common.ELocation.Blob():
+		p, err = createBlobPipeline(ctx, credential)
+	case common.ELocation.File():
+		p, err = createFilePipeline(ctx, credential)
+	case common.ELocation.BlobFS():
+		p, err = createBlobFSPipeline(ctx, credential)
+	default:
+		err = fmt.Errorf("can't produce new pipeline for location %s", location)
+	}
+
+	return
+}
+
+func trimWildcards(path string) string {
+	if strings.Index(path, "*") == -1 {
+		return path
+	}
+
+	firstWCIndex := strings.Index(path, "*")
+	result := replacePathSeparators(path[:firstWCIndex])
+	lastSepIndex := strings.LastIndex(result, "/")
+	result = result[:lastSepIndex+1]
+
+	return result
+}
+
+func (cca *cookedCopyCmdArgs) findObject(source bool, dstIsDir bool, object storedObject) (relativePath string) {
```

Comment:
This feels out of place as a method of the cooked command _args_.  I think if you follow Ze's suggestion above, about moving code out of the big switch statement, you could move it into somewhere/someplace where this could be move to too. (e.g. some kind of copyConfigurator/copyInitiator or something. Can't quite think what the best name would be)

Reply:
My feeling is that a lot of these things feel out of place _everywhere_ as is. copyUtilHandler is bloated, cooked command args isn't the right place for these things, etc.

I also don't want to go causing another copyHandlerUtil style incident.